src/site/stages/build/tome-sync-page-builder.js

```diff
@@ -0,0 +1,159 @@
+ * properties that we want to keep for this entity.
+ *
+ * @param {String} entityType - The type of entity; corresponds to the
+ *                              name of the file.
+ * @param {Object} entity - The contents of the entity itself before
+ *                          reference expansion.
+ *
+ * @return {Array<String>} - A list of properties of the entity to
+ *                           keep.
+ */
+const getEntityProperties = (entityType, entity) =>
+  // TODO: Change this to check the entity type and subtype and return
+  // a customized list.
+  Object.keys(entity);
+
+/**
+ * Takes an entity type and uuid, reads the corresponding file,
+ * searches for references to other entities, and replaces the
+ * references with the contents of those entities recursively.
+ *
+ * @param {String} entityType - The type of entity; corresponds to the
+ *                              name of the file.
+ * @param {String} uuid - The UUID of the entity; corresponds to the
+ *                        name of the file.
+ * @param {Array<String>} parents - A list of parent entities; used to
+ *                                  avoid circular references.
+ *
+ * @return {Object} - The entity with all the references filled in with
+ *                    the body of the referenced entities.
+ */
+const assembler = (entityType, uuid, parents = []) => {
+  // Avoid circular references
+  if (parents.includes(id(entityType, uuid))) {
+    /* eslint-disable no-console */
+    console.log(`I'm my own grandpa! (${id(entityType, uuid)})`);
+    console.log(`  Parents:\n    ${parents.join('\n    ')}`);
+    /* eslint-enable no-console */
+
+    // If we find a circular references, it needs to be addressed;
+    // just quit
+    process.exit(1);
+  }
+
+  const entity = fetchEntity(entityType, uuid);
+
+  // TODO: Once we get the list of properties, the object we return
+  // should have _only_ those properties.
+  const propList = getEntityProperties(entityType, entity);
+
+  // Iterate over all non-blacklisted properties in an entity, look
+  // for references to other identities recursively, and replace the
+  // reference with the entity contents.
+  for (const key of propList) {
+    // eslint-disable-next-line no-continue
+    if (blackList.has(key)) continue;
+    const prop = entity[key];
+
+    // Properties with target_uuids are always arrays from tome-sync
+    if (Array.isArray(prop)) {
+      prop.forEach((item, index) => {
+        const { target_uuid: targetUuid, target_type: targetType } = item;
+
+        // We found a reference! Override it with the expanded entity.
+        if (targetUuid && targetType) {
+          entity[key][index] = assembler(
+            targetType,
+            targetUuid,
+            parents.concat([id(entityType, uuid)]),
+          );
+        }
+      });
+    }
+  }
+
+  return entity;
+};
+
+/**
+ * Get all the starting nodes
+ */
+const getAllNodes = () =>
```

Comment:
`getAllNodes()` -> query something in memory from an object
`readAllNodes()` -> read a file 
`fetch()` -> perform a network request 

^ standardizing the nomenclature makes this easier to read. not sure what's best for something like this that is reading and filtering.